README_local_MacOS.md

```diff
@@ -0,0 +1,122 @@
+# Example Local Dev Environment Setup
```

Comment:
Suggestion: name the file `BUILD_macos.md` ?

Reply:
Sure.

Reply:
I'll rename it once all the other reviews are done.

Reply:
Renamed